EIPS/eip-6366.md

```diff
@@ -33,113 +34,26 @@ Compliant contracts MUST implement `IEIP6366Core`.
-
-```solidity
-function hasPermission(address _owner, address _actor, uint256 _required) external view returns (bool isPermissioned);
-```
-
-#### delegated
-
-Returns the subset permission of the `_owner` address were granted to `_delegatee` address.
-
-```solidity
-function delegated(address _owner, address _delegatee) external view returns (uint256 permission);
-```
-
-#### Transfer
-
-MUST trigger when permission are transferred, including zero permission transfers.
-
-A token contract which creates new tokens SHOULD emit a `Transfer` event with the `_from` address set to `address(0x00)` when tokens are created.
-
-```solidity
-event Transfer(address indexed _from, address indexed _to, uint256 _value)
-```
-
-#### Approval
-
-MUST trigger on any successful call to `approve(address _delegatee, uint256 _permission)`.
-
-```solidity
-event Approval(address indexed _owner, address indexed _delegatee, uint256 _permission)
-```
+1. It is RECOMMENDED to define each permission as a power of `2` so that we can check for the relationship between sets of permissions using [eip-bit_based_permission](./bit_based_permission.md).
+2. `transfer(address _to, uint256 _permission)` transfers a subset of `_permission` permission to address `_to`, and MUST emit the `Transfer` event. The function SHOULD `revert` if the message caller's account permission does not have the subset of the transferring permission. The function SHOULD `revert` if any of transferring permission is existing on target `_to` address. Transfers of `0` permission MUST be treated as normal transfers and emit the `Transfer` event.
+3. `approve(address _delegatee, uint256 _permission)` allows `_delegatee` to act for the permission owner's behalf, up to the `_permission` permission, and MUST emit the `Approval` event. If this function is called again it overwrites the current granted with `_permission`. `_permission` MUST be a subset of all available permission of permission owner. `approve()` method SHOULD `revert` if granting `_permission` permission is not a subset of all available permission of permission owner.
+4. `permissionOf(address _owner)` returns the account permission of the given `_owner` address.
+5. `permissionRequire(uint256 _required, uint256 _permission)` returns `true` if `_required` permission is a subset of `_permission` permission otherwise return `false`.
+6. `hasPermission(address _owner, address _actor, uint256 _required)` returns `true` if `_required` permission is a subset of `_actor`'s permissions or a subset of his delegated permissions granted by the `_owner`.
+7. `delegated(address _owner, address _delegatee)` returns the subset permission of the `_owner` address were granted to `_delegatee` address.
+8. A token contract which creates new tokens SHOULD emit a `Transfer` event with the `_from` address set to `address(0x00)` when tokens are created.
```

Comment:
To honest I find that this way of defining each function of the interface in a bloc makes the reading too heavy because all the explanations are stuck to each other.
I would prefer something like you can find on others EIP like [1155](https://eips.ethereum.org/EIPS/eip-1155):
![image](https://user-images.githubusercontent.com/43795504/222241881-da67703b-3183-4c55-8c01-a83aa394cc6f.png)